azurelinuxagent/common/cgroups/cgutils.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+
+import re
+
+from azurelinuxagent.common.osutil import get_osutil
+from azurelinuxagent.common.osutil.default import BASE_CGROUPS
+
+WRAPPER_CGROUP_NAME = "WALinuxAgent"
+AGENT_CGROUP_NAME = "WALinuxAgent"
+METRIC_HIERARCHIES = ['cpu', 'memory']
+MEMORY_DEFAULT = -1
+
+# percentage of a single core
+DEFAULT_CPU_LIMIT_AGENT = 10
+DEFAULT_CPU_LIMIT_EXT = 40
+
+DEFAULT_MEM_LIMIT_MIN_MB = 256  # mb, applies to agent and extensions
+DEFAULT_MEM_LIMIT_MAX_MB = 512  # mb, applies to agent only
+DEFAULT_MEM_LIMIT_PCT = 15  # percent, applies to extensions
+
+re_user_system_times = re.compile('user (\d+)\nsystem (\d+)\n')
+
+
+class CGroupsException(Exception):
+
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return repr(self.msg)
+
+
+class Cpu(object):
+    def __init__(self, cgt):
+        """
+        Initialize data collection for the Cpu hierarchy. User must call update() before attempting to get
+        any useful metrics.
+
+        :param cgt: CGroupsTelemetry
+        :return:
+        """
+        self.cgt = cgt
+        self.osutil = get_osutil()
+        self.current_cpu_total = self.get_current_cpu_total()
+        self.previous_cpu_total = 0
```

Comment:
seems like previous is used only internally, let's mark it as private

Reply:
It is used in the `__str__` method, and cannot be made private to the `__init__` method.

Reply:
I mean add "_" to the beginning of the name - naming convention for "private" members